### ISSUE TITLE
No KafkaJS error type definitions

### ISSUE THREAD

**Issue by trevorr** [confluentinc/confluent-kafka-javascript#193]
**Environment Information**

 - confluent-kafka-javascript version: 0.5.2

**Steps to Reproduce**

The migration docs mention using `Kafka.isKafkaJSError` for error handling, but this function is not present in the Typescript typings. None of the `KafkaJSError` classes are present either. I'm seeing `KafkaJSProtocolError` thrown in production with the message `Broker: Unknown topic or partition`, and I would like to handle that, but there doesn't seem to be a clean way to do that from Typescript.

**Comment by milindl** [comment]
The latest release contains these types 

I'd recommending checking the code after making sure it's a KafkaJS error, for example

```
import { KafkaJS } from '@confluentinc/kafka-javascript';
...
// e is the thrown or caught error 
if (!KafkaJS.isKafkaJSError(e)) throw e;
const kErr = e as KafkaJS.KafkaJSError;
if (kErr.code === KafkaJS.ErrorCodes.ERR__TIMED_OUT /* or ERR_UNKNOWN_TOPIC_OR_PART etc. ERR__ indicate 'local' errors raised by the library and ERR_ indicate errors from the broker. */) { 
  /* ... */
}
```

The types for the error codes are present. You could look them up [here ](https://docs.confluent.io/platform/current/clients/confluent-kafka-javascript/docs/RdKafka.html#.LibrdKafkaError.codes)too.

**Comment by trevorr** [comment]
Great, thanks! For anyone following along, this was fixed by #210.
